modules/edu.gemini.giapi.client/src/main/scala/edu/gemini/giapi/client/gpi/GPIClient.scala

```diff
@@ -4,23 +4,25 @@
 package edu.gemini.giapi.client.gpi
 
 import cats.Show
-import cats.instances.string._
+import cats.effect.Effect
 import cats.instances.double._
 import cats.instances.int._
 import cats.syntax.show._
 import edu.gemini.aspen.giapi.commands.{Activity, Command, DefaultConfiguration, SequenceCommand}
-import edu.gemini.giapi.client.{Giapi, commands}
 import edu.gemini.giapi.client.commands.CommandResult
+import edu.gemini.giapi.client.{Giapi, commands}
 import fs2.Stream
 import mouse.boolean._
+import scala.concurrent.ExecutionContext
 
 /**
   * Client for GPI
   */
-// Status items
-class GPIClient[F[_]](giapi: Giapi[F]) {
+class GPIClient[F[_]: Effect](giapi: Giapi[F], ec: ExecutionContext) {
```

Comment:
@tpolecat I'm not that happy having `Effect` here when I use it only for one method but I don't know how to do it otherwise as i need `F` in `Giapi[F]`

Reply:
Which method needs it? In general you can say `def foo(args)(implicit ev: Effect[F])` to ensure that `F` *also* has `Effect` in addition to whatever else you know. This may mean that some higher-level methods will also need to add the constraint.

Reply:
`stream` uses `Effect` as required my `fs2`